### PR TITLE
Updating System.Reflection.Metadata to 1.1.0-alpha-0015 and System.Collections.Immutable to 1.1.37.0

### PR DESCRIPTION
**Pull request by tannergooding**

### CODE REVIEW

File: src/Compilers/CSharp/CscCore/project.lock.json [hunk 1]
```diff
@@ -1074,12 +1084,22 @@
...
+          "System.Globalization": "[4.0.0, )",
+          "System.Linq": "[4.0.0, )",
+          "System.Resources.ResourceManager": "[4.0.0, )",
+          "System.Runtime": "[4.0.0, )",
+          "System.Runtime.Extensions": "[4.0.0, )",
+          "System.Threading": "[4.0.0, )"
+        },
         "compile": {
-          "lib/portable-net45+win8+wp8+wpa81/System.Collections.Immutable.dll": {}
+          "lib/dotnet/System.Collections.Immutable.dll": {}
```
Comment (tmeschter) [inline src/Compilers/CSharp/CscCore/project.lock.json]: What's the significance of this changing from `portable-net45+win8+wp8+wpa81` to `dotnet`?

**Comment by tannergooding** [comment]
test Linux please
test mac please

**Comment by tannergooding** [comment]
test Linux please
test mac please

File: src/Compilers/CSharp/CscCore/project.lock.json (hunk 1 above)
Comment (nguerrera) [inline src/Compilers/CSharp/CscCore/project.lock.json]: There is no real significance. The binaries are identical. The portable-*/ folder exists to support older projects and older versions of nuget.

**Pull request merged**